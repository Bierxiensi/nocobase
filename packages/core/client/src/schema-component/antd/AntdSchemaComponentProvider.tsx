import * as components from '.';
import { Plugin } from '../../application-v2/Plugin';
import * as common from '../common';
import { useFilterActionProps } from './filter/useFilterActionProps';
import { requestChartData } from './g2plot/requestChartData';

export class AntdSchemaComponentPlugin extends Plugin {
  async load() {
    this.addComponents();
    this.addScopes();
  }

  addComponents() {
    this.app.addComponents({
      ...(components as any),
      ...common,
    });
  }

  addScopes() {
    this.app.addScopes({
      requestChartData,
      useFilterActionProps,
    });
  }
}
