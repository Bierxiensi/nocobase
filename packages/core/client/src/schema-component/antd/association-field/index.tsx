import { AssociationField } from './AssociationField';
import { AssociationFieldAddNewer } from './AssociationField.AddNewer';
import { AssociationFieldSelector } from './AssociationField.Selector';
import { AssociationFieldViewer } from './AssociationField.Viewer';
import { AssociationFieldNesterDecorator } from './AssociationField.Nester.Decorator';
import { Nester } from './AssociationField.Nester';
AssociationField.AddNewer = AssociationFieldAddNewer;
AssociationField.Selector = AssociationFieldSelector;
AssociationField.Viewer = AssociationFieldViewer;
AssociationField.Nester = Nester;
AssociationField.Nester.Decorator = AssociationFieldNesterDecorator;

export * from './components/SubForm';
export * from './components/TableField';

export { AssociationField };
