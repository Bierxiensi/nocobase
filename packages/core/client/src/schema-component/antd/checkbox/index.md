---
group:
  title: Schema Components
  order: 3
---

# Checkbox

## Examples

### 勾选

<code src="./demos/demo1.tsx"></code>

### 组

<code src="./demos/demo2.tsx"></code>
