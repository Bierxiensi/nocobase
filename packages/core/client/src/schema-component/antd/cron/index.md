---
nav:
  path: /client
group:
  path: /schema-components
---

# Cron

## Example

#### Cron

<code src="./demos/demo1.tsx"></code>

#### CronSet

<code src="./demos/demo2.tsx"></code>
