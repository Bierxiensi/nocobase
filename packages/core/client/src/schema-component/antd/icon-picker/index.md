---
group:
  title: Schema Components
  order: 3
---

# IconPicker

## Examples

### IconPicker

<code src="./demos/demo1.tsx"></code>
