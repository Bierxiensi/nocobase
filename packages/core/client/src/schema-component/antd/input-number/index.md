---
group:
  title: Schema Components
  order: 3
---

# InputNumber

## Examples

### InputNumber

<code src="./demos/demo1.tsx"></code>

### addonBefore/addonAfter

<code src="./demos/demo2.tsx"></code>

### High precision decimals

<code src="./demos/demo3.tsx"></code>
