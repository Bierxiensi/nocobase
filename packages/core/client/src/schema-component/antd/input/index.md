---
group:
  title: Schema Components
  order: 3
---

# Input

## Nodes

- Input
- Input.TextArea
- Input.URL

## Examples

### Input

<code src="./demos/demo1.tsx"></code>

### Textarea

<code src="./demos/demo2.tsx"></code>

### URL

<code src="./demos/demo3.tsx"></code>

### JSON

<code src="./demos/demo4.tsx"></code>
