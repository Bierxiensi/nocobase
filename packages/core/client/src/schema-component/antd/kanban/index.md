---
group:
  title: Schema Components
  order: 3
---

# Kanban

## 示例

<code src="./demos/demo1.tsx"></code>
