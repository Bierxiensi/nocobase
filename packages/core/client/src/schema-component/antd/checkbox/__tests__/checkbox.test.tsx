import { FormItem } from '@formily/antd';
import React from 'react';
import { Checkbox } from '../Checkbox';
import { SchemaComponent, SchemaComponentProvider } from '../../../../schema-component';
import { render ,fireEvent,screen} from '@testing-library/react';

const schema = {
  type: 'object',
  properties: {
    input: {
      type: 'boolean',
      title: `edit`,
      'x-decorator': 'FormItem',
      'x-component': 'Checkbox',
      'x-reactions': {
        target: 'read',
        fulfill: {
          state: {
            value: '{{$self.value}}',
          },
        },
      },
    },
    read: {
      type: 'boolean',
      title: `阅读模式`,
      'x-read-pretty': true,
      'x-decorator': 'FormItem',
      'x-component': 'Checkbox',
    },
  },
};

const App = () => {
  return (
    <SchemaComponentProvider components={{ Checkbox, FormItem }}>
      <SchemaComponent schema={schema} />
    </SchemaComponentProvider>
  );
};

describe('checkbox', () => {
  it('checkbox', async() => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
    const checkbox=container.getElementsByClassName('ant-checkbox-wrapper')[0]
    fireEvent.click(checkbox);
     expect(checkbox).toHaveClass('ant-checkbox-wrapper-checked');
  });
});
