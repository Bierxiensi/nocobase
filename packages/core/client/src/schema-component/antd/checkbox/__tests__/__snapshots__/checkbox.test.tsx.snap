// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox checkbox 1`] = `
<div>
  <div
    class="ant-formily-item ant-formily-item-layout-horizontal ant-formily-item-feedback-layout-loose ant-formily-item-label-align-right ant-formily-item-control-align-left"
  >
    <div
      class="ant-formily-item-label"
    >
      <div
        class="ant-formily-item-label-content"
      >
        <span>
          <label>
            edit
          </label>
        </span>
      </div>
      <span
        class="ant-formily-item-colon"
      >
        :
      </span>
    </div>
    <div
      class="ant-formily-item-control"
    >
      <div
        class="ant-formily-item-control-content"
      >
        <div
          class="ant-formily-item-control-content-component"
        >
          <label
            class="ant-checkbox-wrapper"
          >
            <span
              class="ant-checkbox"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
                value=""
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-formily-item ant-formily-item-layout-horizontal ant-formily-item-feedback-layout-loose ant-formily-item-label-align-right ant-formily-item-control-align-left"
  >
    <div
      class="ant-formily-item-label"
    >
      <div
        class="ant-formily-item-label-content"
      >
        <span>
          <label>
            阅读模式
          </label>
        </span>
      </div>
      <span
        class="ant-formily-item-colon"
      >
        :
      </span>
    </div>
    <div
      class="ant-formily-item-control"
    >
      <div
        class="ant-formily-item-control-content"
      >
        <div
          class="ant-formily-item-control-content-component"
        />
      </div>
    </div>
  </div>
</div>
`;
